ray-curator/ray_curator/stages/modules/filter.py

```diff
@@ -0,0 +1,500 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from collections.abc import Callable
+from dataclasses import dataclass
+
+import fasttext
+import huggingface_hub
+import pandas as pd
+from transformers import AutoTokenizer
+
+from ray_curator.backends.base import NodeInfo, WorkerMetadata
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.filters.doc_filter import DocumentFilter
+from ray_curator.tasks import DocumentBatch
+
+
+@dataclass
+class Score(ProcessingStage[DocumentBatch, DocumentBatch]):
+    """
+    The module responsible for adding metadata to records based on statistics about the text.
+    It accepts an arbitrary scoring function that accepts a text field and returns a score.
+    It also accepts a DocumentFilter object, in which case the score_fn will be the score_document method of the DocumentFilter.
+
+    Unlike ScoreFilter, it does not filter based on the computed score.
+    It only adds metadata to the record.
+
+    Args:
+        score_fn (Callable | DocumentFilter): The score function or the DocumentFilter object. If it is a DocumentFilter object, the score_fn will be the score_document method of the DocumentFilter.
+        score_field (str): The field the score will be stored in.
+        text_field (str): The field the documents will be read from.
+        processing_batch_size (int): The number of tasks to process in a batch.
+
+    """
+
+    score_fn: Callable | DocumentFilter
+    score_field: str
+    text_field: str = "text"
+    processing_batch_size: int = 1
+
+    @property
+    def name(self) -> str:
+        return self.score_fn.name if isinstance(self.score_fn, DocumentFilter) else "score_fn"
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.text_field]
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.score_field]
+
+    @property
+    def batch_size(self) -> int:
+        """Number of tasks to process in a batch."""
+        return self.processing_batch_size
+
+    def setup_on_node(self, _node_info: NodeInfo, _worker_metadata: WorkerMetadata) -> None:
+        if self.name in ["lang_id", "fasttext_quality_filter"]:
```

Comment:
Seems like the code path below is only when `score_fn` is a DocumentFilter if so can you add line above that says `if isinstance(score_fn, Callable): continue`

edit ignore this, relevant comment below https://github.com/NVIDIA-NeMo/Curator/pull/746/files/95e8e039f3bbee4454b04cddad97b74ca7a5f4be#r2178715264